src/setup-java.ts

```diff
@@ -19,12 +20,50 @@ async function run() {
     const checkLatest = getBooleanInput(constants.INPUT_CHECK_LATEST, false);
     let toolchainIds = core.getMultilineInput(constants.INPUT_MVN_TOOLCHAIN_ID);
 
+    core.startGroup('Installed distributions');
+
     if (versions.length !== toolchainIds.length) {
       toolchainIds = [];
     }
 
-    core.startGroup('Installed distributions');
+    if (!versions.length) {
+      core.debug('JAVA_VERSION input is empty, looking for .java-version file');
+      const versionFileName = '.java-version';
+      const contents = fs
+        .readFileSync(versionFileName)
+        .toString()
+        .trim();
+      const semverRegExp = /(\d+\.\d+\.\d+.*ea.*$|\d+\.\d+\.\d+\+\d*$|\d+\.\d+\.\d+|\d+\.\d+|\d+\-ea$|\d+$)/;
+      let version = semverRegExp.test(contents) ? RegExp.$1 : '';
```

Comment:
```suggestion
      const semverRegExp = /(?<version>\d+\S*)(\s|$)/m;
      let version = contents.match(semverRegExp)?.groups?,version;
```
I suggest simplifying the RegExp, as we just need to strip distribution name optionally. The RegExp.$1-9 syntax is deprecated we can use something newer. The suggested above RegExp is subject to testing.

Reply:
> it`s not gonna work as expected ![image](https://user-images.githubusercontent.com/102794661/204349816-b55859fc-327e-4247-84c5-7f8884b872ca.png)

Try to use this one: `/(?<version>(?<=^|\s|\-)(\d+\S*))(\s|$)/m`